File: extensions/panache/hibernate-orm-panache-kotlin/deployment/src/test/java/io/quarkus/hibernate/orm/panache/kotlin/deployment/test/DefaultPanacheEntityToStringTest.java
package io.quarkus.hibernate.orm.panache.kotlin.deployment.test;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntity;

public class DefaultPanacheEntityToStringTest {

    @Test
    public void testDefaultToStringMethod() {
        MyPanacheEntity myPanacheEntityWithId = new MyPanacheEntity(2912l);
        assertEquals("MyPanacheEntity<2912>", myPanacheEntityWithId.toString());

        MyPanacheEntity myPanacheEntityWithNullId = new MyPanacheEntity(null);
        assertEquals("MyPanacheEntity<null>", myPanacheEntityWithNullId.toString());
    }

    static class MyPanacheEntity extends PanacheEntity {
        public MyPanacheEntity(Long id) {
            this.setId(id);
        }
    }
}

File: extensions/panache/hibernate-orm-panache-kotlin/deployment/src/test/kotlin/io/quarkus/hibernate/orm/panache/kotlin/deployment/test/DefaultPanacheEntityToStringTest.kt
package io.quarkus.hibernate.orm.panache.kotlin.deployment.test

import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntity
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class DefaultPanacheEntityToStringTest {
    @Test
    fun testDefaultToStringMethod() {
        val myPanacheEntityWithId = MyPanacheEntity(2912L)
        Assertions.assertEquals("MyPanacheEntity<2912>", myPanacheEntityWithId.toString())
        val myPanacheEntityWithNullId = MyPanacheEntity(null)
        Assertions.assertEquals("MyPanacheEntity<null>", myPanacheEntityWithNullId.toString())
    }

    internal class MyPanacheEntity(id: Long?) : PanacheEntity() {
        init {
            this.id = id
        }
    }
}
